helm-chart/images/binderhub/requirements.in

```diff
@@ -1,8 +1,10 @@
-# The dependencies in this file, together with binderhub's dependencies, can be
-# used to update requirements.txt for this Docker image:
-#
-#    ./dependencies freeze --upgrade
-#
+# make sure to run in the same Python version and OS as the Dockerfile,
+# e.g.
+
+#    docker run --rm -it -e CUSTOM_COMPILE_COMMAND="see requirements.in" -v$PWD:/io -w /io/helm-chart/images/binderhub python:3.8 sh -c 'pip install pip-tools==6.* && pip-compile'
```

Comment:
If desired, we can keep the `dependencies` script as a callable shortcut to this command, inserting `sys.argv` after `pip-compile`.

Reply:
Would you mind adding some of the `How it works:` section from the old dependencies script here? I think it includes some useful context for why we're doing this.

Reply:
it started to get a bit long for a comment, so I added a README with some explanations